tools/astarte_export/lib/fetchdata/queries/queries.ex

```diff
@@ -0,0 +1,249 @@
+defmodule Astarte.Export.FetchData.Queries do
+  alias Astarte.Core.InterfaceDescriptor
+  alias Astarte.Core.Mapping
+  require Logger
+
+  def get_connection() do
+    nodes = Application.get_env(:xandra, :cassandra_nodes)
+    {host, port} = Enum.random(nodes)
+    Logger.info("Connecting to #{inspect(host)}:#{inspect(port)} cassandra database.")
+
+    with {:ok, xandra_conn} <- Xandra.start_link(nodes: ["#{host}:#{port}"], atom_keys: true) do
+      Logger.info("Connected to database.")
+      {:ok, xandra_conn}
+    else
+      {:error, reason} ->
+        Logger.error("DB connection setup failed: #{inspect(reason)}", tag: "db_connection_failed")
+    end
+  end
+
+  def retrieve_interface_row(conn, realm, interface, major_version, options) do
+    interface_statement = """
+    SELECT name, major_version, minor_version, interface_id, type, ownership, aggregation,
+      storage, storage_type, automaton_transitions, automaton_accepting_states
+    FROM #{realm}.interfaces
+    WHERE name=? AND major_version=?
+    """
+
+    params = [{"ascii", interface}, {"int", major_version}]
+
+    page_options = get_page_options(options)
+
+    options = [uuid_format: :binary, timestamp_format: :integer] ++ page_options
+
+    with {:ok, result} <- Xandra.execute(conn, interface_statement, params, options) do
+      {:ok, result}
+    else
+      {:error, %Xandra.Error{message: message}} ->
+        Logger.error("database error: #{inspect(message)}.", tag: "database_error")
+        {:error, :database_error}
+
+      {:error, %Xandra.ConnectionError{} = err} ->
+        Logger.error("database connection error:#{inspect(err)}.",
+          tag: "database_connection_error"
+        )
+
+        {:error, :database_connection_error}
+    end
+  end
+
+  def fetch_interface_descriptor(conn, realm, interface, major_version, options) do
```

Comment:
Did you copy this code from astarte_data_access?

Reply:
I had looked into Astarte.Import and used the same. Currently Astarte.Import uses Xandra interface.

Moreover as per the documentation, these are the 2 errors the Xandra interface supports
**Documentation says**: 
Errors returned by this function can be either Xandra.Error or Xandra.ConnectionError structs. See the module documentation for more information about errors.

Reply:
astarte_data_access implemented this function using CQex. i had updated the same with Xandra, as we are migrating from CQex to Xandra